doc/basic_types.md

```diff
@@ -1,5 +1,360 @@
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+Node Addon API consists of a few fundamental data types. These allow a user of 
+the API to create, convert and introspect fundamental JavaScript types, and
+interop with their C++ counterparts.
+
+## Value
+
+Value is the base class of Node Addon API's fundamental object type hierarchy.
+It represents a JavaScript value of an unknown type. It is a thin wrapper around
+the N-API datatype `napi_value`. Methods on this class can be used to check
+the JavaScript type of the underlying `napi_value` and also convert to C++ 
+types.
+
+### Constructor
+
+```cpp
+Value();                               
+```
+
+Used to create a Node Addon API `Value` that represents an **empty** value.
+
+```cpp
+Value(napi_env env, napi_value value); 
+```
+
+- `[in] env` - The `napi_env` environment in which to construct the Value object.
+- `[in] value` - The underlying JavaScript value that the `Value` instance represents
+
+Used to create a Node addon API `Value` that represents the `napi_value` passed
+in.
+
+### Operators
+
+#### operator napi_value
+
+```cpp
+operator napi_value() const;
+```
+
+Returns the underlying N-API value primitive. If the instance is _empty_,
+this returns  `nullptr`.
+
+#### operator ==
+
+```cpp
+bool operator ==(const Value& other) const;
+```
+
+Returns `true` if this value strictly equals another value, or `false` otherwise
+
+#### operator !=
+
+```cpp
+bool operator !=(const Value& other) const;
+```
+
+Returns `false` if this value strictly equals another value, or `true` otherwise
+
+### Methods
+
+#### From
+```cpp
+template <typename T>
+static Value From(napi_env env, const T& value);
+```
+
+- `[in] env` - The `napi_env` environment in which to construct the Value object.
+- `[in] value` - The C++ type to represent in JavaScript.
+
+This method is used to convert from a C++ type to a JavaScript value.
+Here, `value` may be any of:
+- `bool` - returns a `Napi::Boolean`
+- Any integer type - returns a `Napi::Number`
+- Any floating point type - returns a `Napi::Number`
+- `const char*` (encoded using UTF-8, null-terminated) - returns a `Napi::String`
+- `const char16_t*` (encoded using UTF-16-LE, null-terminated) - returns a `Napi::String`
+- `std::string` (encoded using UTF-8) - returns a `Napi::String`
+- `std::u16string` - returns a `Napi::String`
+- `napi::Value` - returns a `Napi::Value`
+- `napi_value` - returns a `Napi::Value`
+
+#### As
+```cpp
+template <typename T> T As() const;
+```
+
+Casts to another type of `Napi::Value`, when the actual type is known or assumed.
+
+This conversion does NOT coerce the type. Calling any methods inappropriate for 
+the actual value type will throw `Napi::Error`.
+
+#### StrictEquals
+```cpp
+bool StrictEquals(const Value& other) const;
+```
+
+- `[in] other` - The value to compare against
+
+Tests if this value strictly equals another value.
```

Comment:
Maybe it's better move `Tests if this value strictly equals another value.` on top and here something like this: Returns `true` if the value isn strictly equals another value, or `false` otherwise.